agent/connect/ca/provider.go

```diff
@@ -73,4 +74,12 @@ type Provider interface {
 	// is shut down permanently, such as removing a temporary PKI backend in Vault
 	// created for an intermediate CA.
 	Cleanup() error
+
+	// SupportsCrossSigning indicates whether the provider supports cross-signing
+	// other CA certs via CrossSignCA().
+	SupportsCrossSigning() bool
+
+	// MinLifetime returns the minimum TTL allowed by the provider for certificates
+	// it issues.
+	MinLifetime() time.Duration
```

Comment:
Should we have a max too? Might make sense although less likely to be a problem in practice. If it's a lot of work we can leave it for now.

Reply:
I would like to kick this down the road for now. I'm not sure how useful `MaxLifetime` would be, since we want to encourage people to keep the TTL short-lived anyway, don't we? And I haven't yet seen a CA that wouldn't support at least a one-year TTL, which is way more than we want.